validator_test.go

```diff
@@ -60,137 +62,287 @@ func (vt *ValidatorTest) WriteEmails(t *testing.T, emails []string) {
 }
 
 func TestValidatorEmpty(t *testing.T) {
+	testCases := []struct {
```

Comment:
Looking at this file, it seems as though you converted a bunch of standalone tests into table based. The differences seem to just be the input to `WriteEmails` and `NewValidator`, we could probably make this all one table and have all the test cases run from a single test function, WDYT?

Reply:
@JoelSpeed yeah, make sense to have a single test function. 
So, I've bundled up together the test cases in a single function, except the test cases that checks for the dynamic change in the email list.